tools/lou_checkyaml.c

```diff
@@ -282,6 +282,82 @@ read_mode(yaml_parser_t *parser) {
 }
 
 int *
+read_inPos(yaml_parser_t *parser, int len) {
+	int *pos = malloc(sizeof(int) * len);
+	int i = 0;
+	yaml_event_t event;
+	int parse_error = 1;
+	char *tail;
+
+	if (!yaml_parser_parse(parser, &event) || (event.type != YAML_SEQUENCE_START_EVENT))
+		yaml_error(YAML_SEQUENCE_START_EVENT, &event);
+	yaml_event_delete(&event);
+
+	while ((parse_error = yaml_parser_parse(parser, &event)) &&
+			(event.type == YAML_SCALAR_EVENT)) {
+		errno = 0;
+		int val = strtol((const char *)event.data.scalar.value, &tail, 0);
+		if (errno != 0)
+			error_at_line(EXIT_FAILURE, 0, file_name, event.start_mark.line + 1,
+					"Not a valid inpos position '%s'. Must be a number\n",
+					event.data.scalar.value);
+		if ((const char *)event.data.scalar.value == tail)
+			error_at_line(EXIT_FAILURE, 0, file_name, event.start_mark.line + 1,
+					"No digits found in inpos position '%s'. Must be a number\n",
+					event.data.scalar.value);
+		pos[i++] = val;
+		yaml_event_delete(&event);
+	}
+	if (i != len)
+		error_at_line(EXIT_FAILURE, 0, file_name, event.start_mark.line + 1,
+				"Too many or too few inpos positions (%i) for translation of length %i\n", i,
+				len);
+	if (!parse_error) yaml_parse_error(parser);
+	if (event.type != YAML_SEQUENCE_END_EVENT)
+		yaml_error(YAML_SEQUENCE_END_EVENT, &event);
+	yaml_event_delete(&event);
+	return pos;
+}
+
+int *
+read_outPos(yaml_parser_t *parser, int len) {
+	int *pos = malloc(sizeof(int) * len);
+	int i = 0;
+	yaml_event_t event;
+	int parse_error = 1;
+	char *tail;
+
+	if (!yaml_parser_parse(parser, &event) || (event.type != YAML_SEQUENCE_START_EVENT))
+		yaml_error(YAML_SEQUENCE_START_EVENT, &event);
+	yaml_event_delete(&event);
+
+	while ((parse_error = yaml_parser_parse(parser, &event)) &&
+			(event.type == YAML_SCALAR_EVENT)) {
+		errno = 0;
+		int val = strtol((const char *)event.data.scalar.value, &tail, 0);
+		if (errno != 0)
+			error_at_line(EXIT_FAILURE, 0, file_name, event.start_mark.line + 1,
+					"Not a valid outpos position '%s'. Must be a number\n",
+					event.data.scalar.value);
+		if ((const char *)event.data.scalar.value == tail)
+			error_at_line(EXIT_FAILURE, 0, file_name, event.start_mark.line + 1,
+					"No digits found in outpos position '%s'. Must be a number\n",
```

Comment:
I think this code path indicates that chars other than digits where found in the value, so there could be digits but there definitely are also other non-digits characters. Maybe adapt the error message.

Reply:
Forget what I said here. I think the error message is good enough